xbmc/video/guilib/VideoVersionHelper.cpp

```diff
@@ -10,15 +10,137 @@
 
 #include "FileItem.h"
 #include "ServiceBroker.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
 #include "settings/Settings.h"
 #include "settings/SettingsComponent.h"
 #include "settings/lib/Setting.h"
+#include "utils/StringUtils.h"
 #include "utils/log.h"
 #include "video/VideoDatabase.h"
-#include "video/dialogs/GUIDialogVideoVersion.h"
+#include "video/VideoThumbLoader.h"
 
 using namespace VIDEO::GUILIB;
 
+namespace
+{
+struct DialogSelectResult
+{
+  bool cancelled{false};
+  bool button{false};
+  std::shared_ptr<CFileItem> selected;
+};
+
+DialogSelectResult Choose(CGUIDialogSelect& dialog,
+                          const std::string& heading,
+                          const std::string& button,
+                          const CFileItemList& items)
+{
+  CVideoThumbLoader thumbLoader;
+  for (auto& item : items)
+  {
+    thumbLoader.LoadItem(item.get());
+    item->SetLabel2(item->GetVideoInfoTag()->m_strFileNameAndPath);
+  }
+
+  dialog.Reset();
+
+  dialog.SetHeading(heading);
+  dialog.EnableButton(true, button);
+  dialog.SetUseDetails(true);
+  dialog.SetMultiSelection(false);
+  dialog.SetItems(items);
+
+  dialog.Open();
+
+  return {!dialog.IsConfirmed(), dialog.IsButtonPressed(), dialog.GetSelectedFileItem()};
+}
+
+DialogSelectResult ChooseVideoVersion(const std::shared_ptr<CFileItem>& item)
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_VERSION)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_VERSION dialog instance!");
+    return {true, false, {}};
+  }
+
+  CVideoDatabase db;
+  if (!db.Open())
+  {
+    CLog::LogF(LOGERROR, "Unable to open video database!");
+    return {true, false, {}};
```

Comment:
Is this type of return actually needed? Looks ugly as hell. Any way it can be improved? I assume cancelled wont return an item so making it return an optional<item> may be an option. Not sure what button is but maybe returning a dummy file item with some property set could be an alternative?

Reply:
I thought such purely internal stuff does not need to be fancy, just needs to work. But yes, maybe that was selfish and just due to laziness. I will see what I can do about it. ;-)